File: dotnet/SDL.DXA.Modules.CampaignContent/Models/TaggedImage.cs
using Sdl.Web.Common.Models;

namespace SDL.DXA.Modules.CampaignContent.Models
{
    [SemanticEntity(EntityName = "TaggedImage", Vocab = CoreVocabulary, Prefix = "e")]
    public class TaggedImage : EntityModel
    {
        [SemanticProperty("e:name")]
        public string Name { get; set; } 

        [SemanticProperty("e:image")]
        public MediaItem Image { get; set; }

        [SemanticProperty("e:imageUrl")]
        public string ImageUrl { get; set; }

        [SemanticProperty("e:parameters")]
        public string Parameters { get; set; }

        [SemanticProperty("e:altText")]
        public string AltText { get; set; }
    }
}

File: java/campaigncontent-dxa-module/src/main/java/com/sdl/dxa/modules/campaigncontent/model/TaggedImage.java
package com.sdl.dxa.modules.campaigncontent.model;

import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;
import com.sdl.webapp.common.api.model.entity.MediaItem;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

@SemanticEntity(entityName = "TaggedImage",  vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class TaggedImage extends AbstractEntityModel {

    @SemanticProperty("e:name")
    private String name;

    @SemanticProperty("e:image")
    private MediaItem image;

    public String getName() {
        return name;
    }

    public MediaItem getImage() {
        return image;
    }
}
